sqlitecluster/SQLiteNode.cpp

```diff
@@ -2744,12 +2744,30 @@ SQLitePeer* SQLiteNode::_getPeerByID(uint64_t id) const {
     }
 }
 
-SQLitePeer* SQLiteNode::_getPeerByName(const string& name) const {
+bool comparePeers(SQLitePeer peer1, SQLitePeer peer2) {
+    return peer1.name < peer2.name;
+}
+
+int SQLiteNode::binarySearchPeers(const string& name) {
```

Comment:
use `std::binary_search`

Reply:
`std::binary_search` returns true if an item is found or not, it doesn't indicate the position

Reply:
This is probably going to turn out to be significantly easier by just turning `_peerList` into a `set`.